src/compiler/transformers/module/module.ts

```diff
@@ -430,26 +430,32 @@ namespace ts {
          */
         function addExportEqualsIfNeeded(statements: Statement[], emitAsReturn: boolean) {
             if (currentModuleInfo.exportEquals) {
-                if (emitAsReturn) {
-                    const statement = createReturn(currentModuleInfo.exportEquals.expression);
-                    setTextRange(statement, currentModuleInfo.exportEquals);
-                    setEmitFlags(statement, EmitFlags.NoTokenSourceMaps | EmitFlags.NoComments);
-                    statements.push(statement);
-                }
-                else {
-                    const statement = createStatement(
-                        createAssignment(
-                            createPropertyAccess(
-                                createIdentifier("module"),
-                                "exports"
-                            ),
-                            currentModuleInfo.exportEquals.expression
-                        )
-                    );
+                const expressionResult = importCallExpressionVisitor(currentModuleInfo.exportEquals.expression);

```

Comment:
This should have been `visitNode(currentModuleInfo.exportEquals.expression, importCallExpressionVisitor, isExpression)` as it does the necessary validation you're doing on 435-437 for you.